relayer/workers/beefyrelayer/beefy-ethereum-writer.go

```diff
@@ -99,14 +101,30 @@ func (wr *BeefyEthereumWriter) WriteNewSignatureCommitment(ctx context.Context,
 		return fmt.Errorf("Error converting BeefyRelayInfo to BeefyJustification: %s", err.Error())
 	}
 
-	msg, err := beefyJustification.BuildNewSignatureCommitmentMessage(valIndex)
+	contract := wr.lightClientBridge
+	if contract == nil {
+		return fmt.Errorf("Unknown contract")
+	}
+
+	signedValidators := []*big.Int{}
+	for i := range beefyJustification.SignedCommitment.Signatures {
+		// TODO: skip over empty/missing signatures
+		// if signature.Option.IsSome() {
```

Comment:
@philipstanislaus not sure, this could be a GSRPC decode bug or potentially a bug in the beefy code that generates this data onchain, but when decoding, i’m getting entries that have a correct value/signature but that have the Option bool set to false, ie, signature.Option.IsSome() is always false

Reply:
@musnit I'll have a look at that tomorrow, but sounds like it's an issue with Beefy.

Reply:
cool, created a task